tests/upgrade/test_crossgrade.sh

```diff
@@ -227,9 +234,38 @@ installIstioSystemAtVersionHelmTemplate() {
     --set-string global.tag="${2}" \
     --set global.defaultPodDisruptionBudget.enabled=true > "${ISTIO_ROOT}/istio.yaml" || die "helm template failed"
 
+    withRetries 3 60 kubectl apply -f "${ISTIO_ROOT}"/istio.yaml
+}
 
+installIstioAtVersionUsingIstioctl(){
+  writeMsg "istioctl install istio using version ${2} from ${3}."
+  istioctl_path="${3}"/bin
+  withRetries 3 60 "${istioctl_path}"/istioctl experimental manifest apply --yes
+}
 
-    withRetries 3 60 kubectl apply -f "${ISTIO_ROOT}"/istio.yaml
+# istioctl x upgrade supports upgrade istio release version
+# from 1.3.x to 1.3.y
+# from 1.3.x to 1.4.0
+upgradeIstioAtVersionUsingIstioctl(){
+  writeMsg "istioctl upgrade istio using version ${2} from ${3}."
+  istioctl_path="${3}"/bin
+  withRetries 3 60 "${istioctl_path}"/istioctl experimental manifest upgrade --yes
+}
+
+istioInstallOptions() {
+  if [[ "${INSTALL_OPTIONS}" == "helm" ]];then
+    installIstioAtVersionUsingHelm "${FROM_HUB}" "${FROM_TAG}" "${FROM_PATH}"
+  else
+    installIstioAtVersionUsingIstioctl "${FROM_HUB}" "${FROM_TAG}" "${FROM_PATH}"
+  fi
+}
+
+istioUpgradeOptions(){
+  if [[ "${INSTALL_OPTIONS}" == "helm" ]];then
+    installIstioAtVersionUsingHelm "${TO_HUB}" "${TO_TAG}" "${TO_PATH}"
+  else
+    upgradeIstioAtVersionUsingIstioctl "${TO_HUB}" "${TO_TAG}" "${TO_PATH}"
+  fi
 }
```

Comment:
```bash
istioUpgradeOptions(){
  if [[ "${INSTALL_OPTIONS}" == "helm" ]];then
    installIstioAtVersionUsingHelm "${TO_HUB}" "${TO_TAG}" "${TO_PATH}"
  else if [[ "${INSTALL_OPTIONS}" == "istioctl" ]];then
    upgradeIstioAtVersionUsingIstioctl "${TO_HUB}" "${TO_TAG}" "${TO_PATH}"
  else
    exit 1
  fi
}
```

Reply:
if not an option, should we exit?

Reply:
good catch, we should only allow two cases here.